example.py

```diff
@@ -198,6 +209,221 @@ def print_rpc_device(device: RpcDevice) -> None:
     print(f"Connected: {device.connected}")
 
 
+def save_endpoints(device: BlockDevice | RpcDevice) -> None:
+    """Save device endpoints."""
+    data_raw = {"shelly": device.shelly, "status": device.status}
+
+    if device.gen in BLOCK_GENERATIONS:
+        data_raw.update({"settings": device.settings})
+        data_normalized = _normalize_block_data(data_raw)
+    else:
+        data_raw.update({"config": device.config})
+        data_normalized = _normalize_rpc_data(data_raw)
+
+    gen = device.gen
+    model = device.model
+    name = MODEL_NAMES.get(model, "Unknown")
+    version = device.firmware_version.replace("/", "-")
+    fixture_path = Path("fixtures") / f"gen{gen}_{name}_{model}_{version}.json"
+
+    print(f"Saving fixture to {fixture_path}")
+
+    with open(fixture_path, "wb") as file:
+        file.write(
+            orjson.dumps(
+                data_normalized, option=orjson.OPT_INDENT_2 | orjson.OPT_SORT_KEYS
+            )
+        )
+        file.write(b"\n")
+
+    coap_context.close()
+    ws_context.close()
+    sys.exit()
+
+
+def _replacement_values() -> dict[str, Any]:
+    return {
+        "wifi": "Wifi-Network-Name",
+        "wifi_mac": "11:22:33:44:55:66",
+        "device_mac": "AABBCCDDEEFF",
+        "device_short_mac": "DDEEFF",
+        "device_name": "Test Name",
+        "switch_name": "Switch Test Name",
+        "input_name": "Input Test Name",
+        "script_name": "Script Test Name",
+        "em_name": "Energy Monitor Test Name",
+        "mqtt_server": "mqtt.test.server",
+        "sntp_server": "sntp.test.server",
+        "coiot_peer": "home-assistant.server:8123",
+    }
+
+
+def _normalize_block_data(data: dict[str, Any]) -> dict[str, Any]:
+    """Normalize data for BLOCK devices."""
+    status = data["status"]
+    shelly = data["shelly"]
+    settings = data["settings"]
+    values = _replacement_values()
```

Comment:
This looks like this should be a constant instead

Reply:
Done